pkg/user/user.go

```diff
@@ -0,0 +1,44 @@
+package user
+
+import (
+	"os"
+	"os/user"
+	"runtime"
+	"strconv"
+)
+
+// Current - workaround for docker images with fully static binary and 32bit linux operating systems.
+// For static binaries NSS library will not be a part of the static binary hence user.Current() fails.
+// For 32bit linux CGO is not enabled so it will not provide linux specific codebase.
+func Current() (*user.User, error) {
+	if os.Getenv("DOCKERIMAGE") == "1" {
+		wd, err := os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+		return &user.User{Uid: "0", Gid: "0", Username: "root", Name: "root", HomeDir: wd}, nil
+	}
+	if runtime.GOARCH == "386" && runtime.GOOS == "linux" {
+		return &user.User{
+			Uid:      strconv.Itoa(os.Getuid()),
+			Gid:      strconv.Itoa(os.Getgid()),
+			Username: os.Getenv("USER"),
+			Name:     os.Getenv("USER"),
+			HomeDir:  os.Getenv("HOME"),
+		}, nil
+	}
+	user, e := user.Current()
+	if e != nil {
+		return nil, e
+	}
+	return user, nil
+}
+
+// HomeDir - return current home directory.
+func HomeDir() (string, error) {
+	user, err := Current()
+	if err != nil {
+		return "", err
+	}
```

Comment:
Also look for $HOME as second option. I am also thinking of adding check for DOCKER and using "/" as home, transparently.